index.js

```diff
@@ -8,51 +8,122 @@ const RGBot = {
 
 }
 
-const BossRoomBot = {
+const CharInfo = {
+  
+  type: ["Mage", "Rogue", "Tank", "Archer"],
+  abilities: [[0,1], [0,1,2], [0,1], [0,1,2]],
+  // teamId
+  abilityTargets: [[1,0], [1,1,-1], [1,-1], [1,1,1]]
 
-  abilities: [0, 1, 2],
+}
+
+const BossRoomBot = {
 
   getEnemies: (tickInfo) => {
-    return RGBot.getEntitiesOnTeam(tickInfo, 1);
+    return RGBot.getEntitiesOnTeam(tickInfo, 1).filter(entry => !entry.broken);
+  },
+  
+  getAllies: (tickInfo) => {
+    return RGBot.getEntitiesOnTeam(tickInfo, 0);
+  },
+
+  getEnemy: (tickInfo, id) => {
+    return RGBot.getEntitiesOnTeam(tickInfo, 1).find(entry => entry.id == id);
+  },
+
+  getAlly: (tickInfo, id) => {
+    return RGBot.getEntitiesOnTeam(tickInfo, 0).find(entry => entry.id == id);
+  },
+
+  nearestEnemy: (tickInfo, position) => {
+    return RGBot.getEntitiesOnTeam(tickInfo, 1).filter(entry => !entry.broken).sort((a,b) => MathFunctions.distanceSq(position, a.position) - MathFunctions.distanceSq(position, b.position)).find(() => true);
   },
 
   startAbility: (ability, position, targetId, actionQueue) => {
     const input = {
       skillId: ability,
       targetId: targetId,
-      xPosition: position.x,
-      yPosition: position.y,
-      zPosition: position.z
+      xPosition: position != null ? position.x : null,
+      yPosition: position != null ? position.y : null,
+      zPosition: position != null ? position.z : null
     }
+    console.log(`Using abilility ${ability} on targetId: ${targetId} at position: ${input.xPosition}, ${input.yPosition}, ${input.zPosition}`)
     actionQueue.queue("PerformSkill", input)
   }
-
+//test
 }
 
 class RGValidator {
 
 }
 
-export function configureBot() {
-  console.log("Unity bot configureBot function called !")
+const MathFunctions = {
+  distanceSq: (position1, position2) => {
+    return Math.pow(position2.x - position1.x, 2) + Math.pow(position2.y - position1.y, 2) + Math.pow(position2.z - position1.z, 2);
+  }
+}
+
+let charType = Math.round(Math.random() * 1000000) % 4;
```

Comment:
This seems a little more extra than the standard code below, but doesn't matter too much to me
```suggestion
let charType = Math.floor(Math.random() * 4); // for 0 through 3 inclusive
```

Reply:
potato / potato

I generally go with the modulo solution because it handles uneven distribution of the random algorithm.

The proposed solution does work, but if the random numbers aren't evenly distributed across the range you don't get the variance expected.